### DISCUSSION TITLE
cwd change handling not working with LazyVim

### DISCUSSION THREAD

**Discussion by shahamran** [rmagatti/auto-session#298] [Q&A] [answered]
Hi, I'm enjoying this plugin a lot, thank you.

The README describes a specific behavior when changing directories, which doesn't work for me:

> AutoSession now tracks cwd changes! By default, handling is as follows: DirChangedPre (before the cwd actually changes): - Save the current session - Clear all buffers %bd!. This guarantees buffers don't bleed to the next session. - Clear jumps. Also done so there is no bleeding between sessions. - Run the pre_cwd_changed_hook DirChanged (after the cwd has changed): - Restore session using new cwd - Run the post_cwd_changed_hook
> 
> Now when the user changes the cwd with :cd some/new/dir auto-session handles it gracefully, saving the current session so there aren't losses and loading the session for the upcoming cwd if it exists.

The behavior I'm facing is:

* Restore an existing session in some dir A, with some open buffers
* cd into dir B
* Buffers from dir A are still open, but cwd has changed

I'm using LazyVim without many extra plugins (in particular, without the "[project]" extra).

How do I approach debugging this? Or maybe I misunderstood the expected behavior?

Thanks.

[project]: <https://www.lazyvim.org/extras/util/project>

**Comment by rmagatti** [comment]
What does your config look like?

**Reply by shahamran** [reply]
lazyvim.json:
```json
{
  "extras": [
    "lazyvim.plugins.extras.coding.mini-surround",
    "lazyvim.plugins.extras.dap.core",
    "lazyvim.plugins.extras.editor.navic",
    "lazyvim.plugins.extras.editor.trouble-v3",
    "lazyvim.plugins.extras.lang.clangd",
    "lazyvim.plugins.extras.lang.json",
    "lazyvim.plugins.extras.lang.markdown",
    "lazyvim.plugins.extras.lang.python",
    "lazyvim.plugins.extras.lang.rust",
    "lazyvim.plugins.extras.ui.mini-indentscope"
  ],
  "news": {
    "NEWS.md": "4964"
  },
  "version": 4
}
```

lua/plugins/custom.lua:
```lua
return {
  -- Disable animation for indent guides
  {
    "echasnovski/mini.indentscope",
    opts = {
      draw = {
        animation = require("mini.indentscope").gen_animation.none(),
      },
    },
  },
  -- Auto-save sessions
  {
    "rmagatti/auto-session",
    opts = {
      log_level = "error",
      auto_session_suppress_dirs = { "~/", "/" },
      auto_session_enable_last_session = true,
      auto_session_root_dir = "~/.local/state/nvim/sessions/",
    },
    keys = {
      {
        "<leader>fs",
        function()
          require("auto-session.session-lens").search_session()
        end,
        desc = "Sessions",
      },
    },
  },
  {
    "nvimdev/dashboard-nvim",
    optional = true,
    opts = function(_, opts)
      local sessions = {
        action = 'lua require("auto-session.session-lens").search_session()',
        desc = " Sessions",
        icon = " ",
        key = "S",
      }

      sessions.desc = sessions.desc .. string.rep(" ", 43 - #sessions.desc)
      sessions.key_format = "  %s"

      table.insert(opts.config.center, 3, sessions)
    end,
  },
  -- Remote config - in noevide
  {
    "amitds1997/remote-nvim.nvim",
    version = "*", -- Pin to GitHub releases
    opts = {
      client_callback = function(port, _)
        local cmd = ("neovide --server localhost:%s"):format(port)
        vim.fn.jobstart(cmd, {
          detach = true,
          on_exit = function(job_id, exit_code, event_type)
            -- This function will be called when the job exits
            print("Job", job_id, "exited with code", exit_code, "Event type:", event_type)
          end,
        })
      end,
    },
  },
  -- Toggle line blame
  {
    "lewis6991/gitsigns.nvim",
    opts = {
      current_line_blame = true,
      current_line_blame_opts = { delay = 200 },
    },
    keys = {
      {
        "<leader>gt",
        function()
          require("gitsigns").toggle_current_line_blame()
        end,
        desc = "Toggle current line blame",
      },
    },
  },
  -- Enable inlay hints by default
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      inlay_hints = {
        enabled = true,
      },
    },
  },
  -- Better terminal
  {
    "akinsho/toggleterm.nvim",
    version = "*",
    opts = {
      open_mapping = "<C-`>",
      insert_mappings = true,
      size = 25,
    },
  },
  -- Documentation generation
  {
    "danymat/neogen",
    config = true,
  },
}
```

no other changes to lazyvim starter template, besides some neovide options and some unrelated keymaps.

**Reply by rmagatti** [reply, marked as answer]
So, you haven't enabled the cwd change feature. It is disabled by default, the readme covers how to enable it. Also when debugging these things it'd useful to set the log level to something like debug and paste in the question or bug report what you've got in your `:mess`

**Reply by shahamran** [reply]
Ah, so I misunderstood the README. When it says:
> By default, handling is as follows: ...

I interpreted it as "it's on by default, and this is what it does", but it should've been: "when it's on, this is what it does by default".

Thanks anyways, and sorry for not including all the relevant info, I'm new to nvim stuff.

**Reply by rmagatti** [reply]
Correct. If you'd like to improve the wording in the readme with a PR that would be most welcome! I agree it's a bit confusing currently! 🙏
